oeps/oep-0031-arch-i18n.rst

```diff
@@ -0,0 +1,270 @@
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2019-01-07
+   * - Review Period
+     -
+
+.. contents::
+   :local:
+   :depth: 3
+
+##########
+Motivation
+##########
+
+Internationalization (i18n) is a vital part of unlocking content for users around the globe and effectively supporting our business partners. In order to help us reuse localized components from app to app, the different parts of the edX codebase should align on a consistent handling of i18n. This document covers our approach to internationalizing new front-end applications and components written in React. This document doesn't necessarily cover internationalization for other technologies, such as the Django templates in edx-platform.
+
+###########
+Terminology
+###########
+
+We will use the term *localized* to refer to a view of a web page that is customized with a user-appropriate language, time zone, number format, and currency. We will use the term *internationalized* to refer to a web page that is designed to have its text, dates, prices, etc. easily swapped out in order to create different localized versions.
+
+In this OEP we use the terms *locale* and *language* more or less interchangeably. However, note that this document mainly addresses making code multilingual, i.e., the issue of presenting the same content in different languages. Considering ways to make it multi-regional, to allow for presenting different content for different countries, is out of scope for this OEP.
+
+*Transifex* is the third-party system that lets us upload messages for translation and download the translations to incorporate them into our software. Except where Transifex-specific concerns are noted, this document will still apply if we change translation systems.
+
+#############
+Specification
+#############
+
+*************************************
+Industry Standards And Best Practices
+*************************************
+
+HTML Considerations
+===================
+
+Set the language of your document with the ``lang`` attribute on your ``html`` tag. Use the UTF-8 encoding, and declare it with a ``meta`` tag in the head of your HTML document. Prefer using characters (``á``) rather than character entities (``&aacute;``) or escapes (``&#225;``).
+
+Grammar and Cultural Considerations
+===================================
+
+Use string interpolation and placeholder variables instead of creating messages by concatenating two or more short messages together. Even if it looks like it's working in English, it won't necessarily be grammatically correct in other languages. This includes uses such as a pagination display: you cannot translate only the word ``of`` and then glue together ``{number} + {translation of "of"} + {number}``. Instead, your string for translation should be ``{page_number} of {total_pages}``, so your translators can move the placeholders into whatever position is required for the grammar rules of their language.
```

Comment:
Looking at the How-To, I'm not seeing this sort of example in there; it's more about how to set up i18n in general, with none of the guidelines and best practices in this document.  

IMHO the example here is pretty clear; I'm sort of inclined to leave this part as-is, since a link to the How-To won't add much more clarity.